app/org/sagebionetworks/bridge/dynamodb/DynamoCompoundActivityDefinitionDao.java

```diff
@@ -0,0 +1,133 @@
+        // query expression
+        DynamoCompoundActivityDefinition ddbHashKey = new DynamoCompoundActivityDefinition();
+        ddbHashKey.setStudyId(studyId.getIdentifier());
+        DynamoDBQueryExpression<DynamoCompoundActivityDefinition> ddbQueryExpr =
+                new DynamoDBQueryExpression<DynamoCompoundActivityDefinition>().withHashKeyValues(ddbHashKey);
+
+        // execute query
+        List<DynamoCompoundActivityDefinition> defList = mapper.query(DynamoCompoundActivityDefinition.class,
+                ddbQueryExpr);
+
+        // because of generics wonkiness, we need to convert this to a list of parent class CompoundActivityDefinition
+        return defList.stream().collect(Collectors.toList());
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public CompoundActivityDefinition getCompoundActivityDefinition(StudyIdentifier studyId, String taskId) {
+        // create key object
+        DynamoCompoundActivityDefinition ddbDef = new DynamoCompoundActivityDefinition();
+        ddbDef.setStudyId(studyId.getIdentifier());
+        ddbDef.setTaskId(taskId);
+
+        // Call DDB mapper. Throw exception if null.
+        DynamoCompoundActivityDefinition loadedDef = mapper.load(ddbDef);
+        if (loadedDef == null) {
+            throw new EntityNotFoundException(CompoundActivityDefinition.class);
+        }
+
+        return loadedDef;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public CompoundActivityDefinition updateCompoundActivityDefinition(StudyIdentifier studyId, String taskId,
+            CompoundActivityDefinition compoundActivityDefinition) {
+        // Currently, both a save expression and a mismatched version number will throw a
+        // ConditionalCheckFailedException, so it's not possible to distinguish between an EntityNotFound and a
+        // ConcurrentModificationException using a single atomic DDB request. Since ConcurrentModification is more
+        // nefarious, we'll use the atomic operation for that and use a load-before-update to check that the def
+        // already exists.
+
+        // Call get() to verify the def exists. This will throw an EntityNotFoundException if it doesn't exist.
+        getCompoundActivityDefinition(studyId, taskId);
+
+        // Currently, all CompoundActivityDefinitions are DynamoCompoundActivityDefinitions.
+        DynamoCompoundActivityDefinition ddbDef = (DynamoCompoundActivityDefinition) compoundActivityDefinition;
+
+        // Set the studyId and taskId. This prevents people from updating the wrong def or updating a def in another
+        // study.
+        ddbDef.setStudyId(studyId.getIdentifier());
+        ddbDef.setTaskId(taskId);
+
+        // Call DDB mapper to save.
+        try {
+            mapper.save(ddbDef);
+        } catch (ConditionalCheckFailedException ex) {
+            throw new ConcurrentModificationException(ddbDef);
+        }
+
+        return ddbDef;
```

Comment:
The version attribute helps us catch race conditions by error'ing them out at the DDB level.

On the other hand, if we take out the version attribute entirely, we can combine the create() and update() methods in the DAO, since at that point, they'd be identical. Let me know if you have a preference either way.